### PR TITLE
feat(inline-property-schema): add new spectral-style rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces the new spectral-style
'inline-property-schema' validation rule.
It will look for and warn about any nested objects that are defined with an inline object schema.

Signed-off-by: Phil Adams <phil_adams@us.ibm.com>

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Dependencies have been updated as needed
- [x] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [ ] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [ ] Removed or adjusted testcases (packages/validator/test)
- [ ] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [ ] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

**Comment by padamstx** [comment]
This PR is stacked up behind the "inline-request-schema" PR.  Once that other PR is merged, I'll adjust this PR's branch to be based off main.

**Comment by padamstx** [comment]
I'm not necessarily a big fan of the name of the new rule (inline-property-schema), mainly because the rule applies to more than just schema properties.  It also checks array "items", "additionalProperties", etc.
Other names I considered were 'inline-nested-object-schema' (too long?), 'inline-nested-schema' (too confusing?), and 'inline-nested-object' (not sure why I didn't pick this one :joy:).

If anyone has any other "name" ideas, let me know.

**Comment by hudlow** [comment]
@padamstx I think `inline-nested-object-schema` is okay, though if I was doing this work from scratch I might suggest we use the word `anonymous` in place of `inline` to indicate more directly why these schemas are a problem.

**Comment by padamstx** [comment]
Potential new name for the rule: `anonymous-nested-object`

If we use "anonymous" in the name for this rule, we should probably consider also changing the names of the other two rules to be: `anonymous-request-object` and `anonymous-response-object`
to be consistent.

Not sure if we even have a deprecation process implemented for our spectral-style ruleset :joy:

File: docs/ibm-cloud-rules.md [hunk 1]
```diff
@@ -2224,6 +2231,81 @@ paths:
...
+<td><b>Rule id:</b></td>
+<td><b>inline-property-schema</b></td>
+</tr>
+<tr>
+<td valign=top><b>Description:</b></td>
+<td>If a nested schema (perhaps a schema property, an array <code>items</code> schema,
+an <code>additionalProperties</code> schema, etc.) is an object with user-defined properties,
+it is a best practice to define the schema as a named schema within the <code>components.schemas</code> section
+of the API definition, and then reference it with a schema $ref instead of defining it as an inline object schema.
+This is a best practice because the SDK generator will use the schema reference when determining the datatype associated
```
Comment (dpopp07) [inline docs/ibm-cloud-rules.md:2246]: I think we (when relevant) include Handbook references also, right? I think this is directly applicable here: https://test.cloud.ibm.com/docs/api-handbook?topic=api-handbook-schemas#nested-object-schemas

**Review by dpopp07 — APPROVED** [review]
Looks good to me!

File: docs/ibm-cloud-rules.md (hunk 1 above)
Comment (padamstx) [reply docs/ibm-cloud-rules.md]: updated it

**Review by pyrooka — APPROVED** [review]
LGTM!

**Pull request merged**